packages/component-library/components/Heading/Heading.tsx

```diff
@@ -0,0 +1,79 @@
+import classNames from "classnames"
+import { createElement } from "react"
+
+const styles = require("./Heading.module.scss")
+
+export type HeadingVariants =
+  | "display-0"
+  | "heading-1"
+  | "heading-2"
+  | "heading-3"
+  | "heading-4"
+  | "heading-5"
+  | "heading-6"
+
+export type AllowedTags =
+  | "pre"
+  | "p"
+  | "a"
```

Comment:
Also, the `div` allowed tag + `children: React.ReactNode` is a loop hole that let's you put anything in there. I think that's OK as it would be pretty unlikely, and at least this makes people thing twice about doing something super weird like using a `script` tag for a heading.

Reply:
Yeah, this will be especially useful when the Paragraph component is built out. You'd be able to use the components in a way that feels closer to using plain ol' html.
```
<Paragraph>This is a paragraph with a <Link href="#">link</Link>.</Paragraph>
<Heading>You can also add <Link href="#">links</Links> within a heading - but why would you want to?</Heading>
```